packages/core/echo/echo-schema/src/database.ts

```diff
@@ -45,7 +41,13 @@ export interface EchoDatabase {
   /**
    * Query objects.
    */
-  query<T extends TypedObject>(filter?: FilterSource<T>, options?: QueryOptions): Query<T>;
+  query(): Query<TypedObject>;
+  query<T extends OpaqueEchoObject = TypedObject>(
+    filter?: Filter<T> | undefined,
+    options?: QueryOptions | undefined,
+  ): Query<T>;
+  query(filter: OperatorFilter<TypedObject>, options?: QueryOptions): Query<TypedObject>;
+  query<T extends {}>(filter?: T | undefined, options?: QueryOptions | undefined): Query<TypedObject>;
```

Comment:
tbh this is technically a breaking change because it forced you to change existing queries for build to pass (e.g. https://github.com/dxos/dxos/pull/5929/files#diff-77617c3cbe23613edc912f0782264e08b4ccde482e5e801cbfe0ab8f4b570f4bR65) . We need to be careful with this sort of thing. It's likely fine but should be communicated before being landed so that people are not surprised when it ships. cc @zhenyasav

Reply:
Right, we're also gonna end up removing untyped filters anyway I think